### PR TITLE
Import 'active_support' before any of its subparts

### LINKED ISSUE #40: Issue with Rails 7.0.1
Testing an upgrade one of our app to Rails 7.0.1
We noticed a crash when starting zhong process (see below for stack trace)

This seems related to this [issue](https://github.com/rails/rails/issues/43851) on Rails.

I've tested manually to add a 

`require "active_support"` in zhong.rb and it seems to work, but I'm not sure it's the way to go.

```shell
bundle exec zhong zhong.rb
You can remove `require 'dalli/cas/client'` as this code has been rolled into the standard 'dalli/client'.
/Users/davidbourguignon/.rvm/gems/ruby-3.0.1@drivy-rangers/gems/activesupport-7.0.1/lib/active_support/xml_mini.rb:184:in `current_thread_backend': uninitialized constant ActiveSupport::XmlMini::IsolatedExecutionState (NameError)
	from /Users/davidbourguignon/.rvm/gems/ruby-3.0.1@drivy-rangers/gems/activesupport-7.0.1/lib/active_support/xml_mini.rb:103:in `backend='
	from /Users/davidbourguignon/.rvm/gems/ruby-3.0.1@drivy-rangers/gems/activesupport-7.0.1/lib/active_support/xml_mini.rb:201:in `<module:ActiveSupport>'
	from /Users/davidbourguignon/.rvm/gems/ruby-3.0.1@drivy-rangers/gems/activesupport-7.0.1/lib/active_support/xml_mini.rb:11:in `<top (required)>'
	from /Users/davidbourguignon/.rvm/gems/ruby-3.0.1@drivy-rangers/gems/activesupport-7.0.1/lib/active_support/core_ext/array/conversions.rb:3:in `require'
	from /Users/davidbourguignon/.rvm/gems/ruby-3.0.1@drivy-rangers/gems/activesupport-7.0.1/lib/active_support/core_ext/array/conversions.rb:3:in `<top (required)>'
	from /Users/davidbourguignon/.rvm/gems/ruby-3.0.1@drivy-rangers/gems/activesupport-7.0.1/lib/active_support/duration.rb:3:in `require'
	from /Users/davidbourguignon/.rvm/gems/ruby-3.0.1@drivy-rangers/gems/activesupport-7.0.1/lib/active_support/duration.rb:3:in `<top (required)>'
	from /Users/davidbourguignon/.rvm/gems/ruby-3.0.1@drivy-rangers/gems/activesupport-7.0.1/lib/active_support/core_ext/time/calculations.rb:3:in `require'
	from /Users/davidbourguignon/.rvm/gems/ruby-3.0.1@drivy-rangers/gems/activesupport-7.0.1/lib/active_support/core_ext/time/calculations.rb:3:in `<top (required)>'
	from /Users/davidbourguignon/.rvm/gems/ruby-3.0.1@drivy-rangers/gems/activesupport-7.0.1/lib/active_support/core_ext/time.rb:4:in `require'
	from /Users/davidbourguignon/.rvm/gems/ruby-3.0.1@drivy-rangers/gems/activesupport-7.0.1/lib/active_support/core_ext/time.rb:4:in `<top (required)>'
	from /Users/davidbourguignon/.rvm/gems/ruby-3.0.1@drivy-rangers/gems/activesupport-7.0.1/lib/active_support/time.rb:12:in `require'
	from /Users/davidbourguignon/.rvm/gems/ruby-3.0.1@drivy-rangers/gems/activesupport-7.0.1/lib/active_support/time.rb:12:in `<top (required)>'
	from /Users/davidbourguignon/.rvm/gems/ruby-3.0.1@drivy-rangers/gems/zhong-0.3.0/lib/zhong.rb:7:in `require'
	from /Users/davidbourguignon/.rvm/gems/ruby-3.0.1@drivy-rangers/gems/zhong-0.3.0/lib/zhong.rb:7:in `<top (required)>'
	from /Users/davidbourguignon/.rvm/gems/ruby-3.0.1@drivy-rangers/gems/zhong-0.3.0/bin/zhong:5:in `require'
	from /Users/davidbourguignon/.rvm/gems/ruby-3.0.1@drivy-rangers/gems/zhong-0.3.0/bin/zhong:5:in `<top (required)>'
	from /Users/davidbourguignon/.rvm/gems/ruby-3.0.1@drivy-rangers/bin/zhong:23:in `load'
	from /Users/davidbourguignon/.rvm/gems/ruby-3.0.1@drivy-rangers/bin/zhong:23:in `<main>'
	from /Users/davidbourguignon/.rvm/gems/ruby-3.0.1@drivy-rangers/bin/ruby_executable_hooks:22:in `eval'
	from /Users/davidbourguignon/.rvm/gems/ruby-3.0.1@drivy-rangers/bin/ruby_executable_hooks:22:in `<main>'
```

### PR DESCRIPTION
**Pull request by godric**
**TL;DR**, Fix compatibility with ActiveSupport >7 which now require "active_support" before anything else

# What?
- Add `require "active_support"` before any other active_support requires

# Why?
- Fixes #40
- This issue has been brought to the attention to Rails maintainers, as it has broken a few gems.
    see [this reply from the maintainers](https://github.com/rails/rails/pull/44450#issuecomment-1042012586)
  > you must require "active_support" before requiring any subparts. That may change in the future, but for now that's the fix you are looking for.
- ActiveSupport since v7 auto-loads (lazy-load) all its constants [here](https://github.com/rails/rails/blob/7c2714eec64ec05409863a0cd123bbd565f2bc5d/activesupport/lib/active_support.rb#L51)

### CODE REVIEW

**Comment by mkonecny** [comment]
@nickelser any chance we get this change in?

**Comment by godric** [comment]
Closing because this does not seem to be an issue anymore in up to date ActiveRecord

**Comment by mlarraz** [comment]
FYI this is an issue on Rails 7.1 again
